### PR TITLE
Fixes #19420 - Add rake task to bulk update download_policy

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: lib/katello/tasks/repository.rake [hunk 1]
```diff
@@ -59,6 +59,25 @@ namespace :katello do
...
+    end
+
+    User.current = User.anonymous_api_admin
+    repos = Katello::Repository.yum_type.where(library_instance_id: nil)
+
+    repos.find_each.with_index do |repo, index|
+      puts "Processing Repository #{index + 1}/#{repos.count}: #{repo.name} (#{repo.id})"
+      ForemanTasks.sync_task(::Actions::Katello::Repository::Update, repo,
+                             download_policy: policy)
+    end
```
Comment (jlsherrill) [inline lib/katello/tasks/repository.rake]: I would recommend adding a blanket rescue here to rescue and print the error with each repo but continue.

Also a test would be nice :)

**Comment by daviddavis** [comment]
ok test

**Comment by daviddavis** [comment]
oops wrong project

[test]

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
@jlsherrill updated. 👍

File: lib/katello/tasks/repository.rake [hunk 2]
```diff
@@ -59,6 +59,29 @@ namespace :katello do
     end
   end
 
+  desc "Change the download policy of all repos. Specify DOWNLOAD_POLICY=policy"
```
Comment (johnpmitsch) [inline lib/katello/tasks/repository.rake]: @daviddavis is it possible to specify the different options for `DOWNLOAD_POLICY` in the description?

**Comment by johnpmitsch** [comment]
works for me! I'll let @jlsherrill give the final approval

File: lib/katello/tasks/repository.rake (hunk 2 above)
Comment (daviddavis) [inline lib/katello/tasks/repository.rake]: Sure

**Comment by daviddavis** [comment]
> It would be nice to change the policy for all repos based on organization

I probably don't have the time for this change but would be happy to hand this off to a Katello dev!

**Comment by johnpmitsch** [comment]
I am not sure if this is related to your PR, but I keep getting this traceback when I run the rake task.  I've migrated, bundled, restarted services, pulled latest... no luck

```
rake aborted!
I18n::InvalidLocale: :en is not a valid locale
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/i18n-0.8.1/lib/i18n.rb:284:in `enforce_available_locales!'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/i18n-0.8.1/lib/i18n.rb:151:in `translate'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/activesupport-4.2.8/lib/active_support/core_ext/array/conversions.rb:68:in `to_sentence'
/home/vagrant/katello/lib/katello/tasks/repository.rake:62:in `block in <top (required)>'
/home/vagrant/katello/lib/katello/tasks/repository.rake:1:in `<top (required)>'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `block in load'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:240:in `load_dependency'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/activesupport-4.2.8/lib/active_support/dependencies.rb:268:in `load'
/home/vagrant/katello/lib/katello/engine.rb:260:in `block (2 levels) in <class:Engine>'
/home/vagrant/katello/lib/katello/engine.rb:259:in `each'
/home/vagrant/katello/lib/katello/engine.rb:259:in `block in <class:Engine>'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:237:in `instance_exec'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:237:in `block in run_tasks_blocks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:245:in `each'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:245:in `each_registered_block'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:237:in `run_tasks_blocks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/engine.rb:657:in `run_tasks_blocks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/application.rb:451:in `block in run_tasks_blocks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/engine/railties.rb:13:in `each'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/engine/railties.rb:13:in `each'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/application.rb:451:in `run_tasks_blocks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/engine.rb:453:in `load_tasks'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:194:in `public_send'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/railties-4.2.8/lib/rails/railtie.rb:194:in `method_missing'
/home/vagrant/foreman/Rakefile:7:in `<top (required)>'
/home/vagrant/.rvm/gems/ruby-2.2.4/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/home/vagrant/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `eval'
/home/vagrant/.rvm/gems/ruby-2.2.4/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)

```

**Comment by johnpmitsch** [comment]
@daviddavis everything works fine, but the description says
```
rake katello:change_download_policy                 # Change the download policy of all repos
```
It seems like it only takes the first sentence? Changing the periods to commas worked to show the full definition.
```
rake katello:change_download_policy                 # Change the download policy of all repos, specify DOWNLOAD_POLICY=policy, options are immediate,on_demand,background
```

**Comment by daviddavis** [comment]
@johnpmitsch that's by design. The `-T` option on rake shows a list of tasks. If you want the full description, you'd run `rake -D katello:change_download_policy`.

**Comment by johnpmitsch** [comment]
@daviddavis I thought that seemed funny, thanks for that tip.

**Comment by daviddavis** [comment]
[test]

**Pull request merged**